app/(dashboard)/[storeId]/(routes)/products/[productId]/components/product-form.tsx

```diff
@@ -18,26 +18,34 @@ import { Input } from "@/components/ui/input"
 import AlertModal from "@/components/modals/alert-modal"
 import ImageUpload from "@/components/ui/image-upload"
 
-interface BillboardFormProps {
-    initialData: Billboard | null;  // we are passing the Store as initialData
+interface ProductFormProps {
+    initialData: Product & {
+        images: Image[]
+    } | null;  // we are passing the Store as initialData
 }
```

Comment:
Why this interface is changed here like this ?

Reply:
The concept used here is called "intersection types" in TypeScript. An intersection type allows you to combine multiple types into a single type. In the specific case of the ProductFormProps interface, it's an intersection of two types:

Product & { images: Image[]; sizeId: string; }: This represents a type that has all the properties of the Product type (from @prisma/client), along with additional properties images (an array of Image objects) and sizeId (a string).

null: This represents the possibility of the initialData being null.

The | (union) operator is used to combine these two types, indicating that initialData can either be of the first type (Product with additional properties) or null. This allows flexibility in the structure of initialData